docs/source/transaction_family_specifications/xo_transaction_family.rst

```diff
@@ -0,0 +1,147 @@
+***********************************
+XO Transaction Family Specification
+***********************************
+
+Overview
+========
+
+The XO transaction family allows users to play a simple board game
+known variously as *tic-tac-toe*, *noughts and crosses*, and *XO*.
+
+State
+=====
+
+An XO state entry consists of the UTF-8 encoding of a string with
+exactly four commas formatted as follows:
+
+``<name>,<board>,<game-state>,<key1>,<key2>``
+
+where
+- <name> is a nonempty string not containing `|`,
+- <board> is a string of length nine containing only `O`, `X`, or `-`
+  (where the first three characters represent the top row of an XO
+  board, the next three represent the middle row, and the last three
+  represent the bottom row), and
+- <game-state> is one of the following: `P1-NEXT`, `P2-NEXT`,
+  `P1-WIN`, `P2-WIN`, or `TIE`.
+
+In the event of a hash collision (i.e. two or more state entries sharing the
+same address), the colliding state entries will stored as the UTF-8
+encoding of the string ``<a-entry>|<b-entry>|...``, where <a-entry>,
+<b-entry>,... are sorted alphabetically.
+
+
+Addressing
+----------
+
+XO data is stored in state using addresses generated from the XO
+family name and the name of the game being stored. In particular, an
+XO address consists of the first six characters of the SHA-512 hash of
+the UTF-8 encoding of the string "xo" (which is "5b7349") plus the
+first 64 characters of the SHA-512 hash of the UTF-8 encoding of the
+game name.
+
+For example, the XO address for a game called "mygame" could be
+generated as follows:
+
+.. code-block:: pycon
+    >>> hashlib.sha512('xo'.encode('utf-8')).hexdigest()[:6] + hashlib.sha512('mygame'.encode('utf-8')).hexdigest()[:64]
+    '5b7349700e158b598043efd6d7610345a75a00b22ac14c9278db53f586179a92b72fbd'
+
+
+Transaction Payload
+===================
+
+An XO transaction request payload consists of the UTF-8 encoding of a
+string with exactly two commas formatted as follows:
+
+``<name>,<action>,<space>``
+
+where
+- <name> is a nonempty string not containing `|`,
+- <action> is either `TAKE` or `CREATE`, and
+- <space> is an integer between one and nine (inclusive) if <action> is `TAKE`.
+
+
+Transaction Header
+==================
+
+Inputs and Outputs
+------------------
+
+The inputs and outputs for an XO transaction are both just the address
+of the transaction's game name
+
+
+Dependencies
+------------
+
+XO transactions have no explicit dependencies.
+
+
+Family
+------
+
+- family_name: "xo"
+- family_version: "1.0"
+
+
+Encoding
+--------
+
+- payload_encoding: "csv-utf8"
+
+Execution
```

Comment:
Consider reformatting this whole section as an ordered list (see the Burrow-EVM spec) for clarity and readability